pep-0632.rst

```diff
@@ -0,0 +1,169 @@
+monkeypatching, but has now taken a copy of the underlying code. [3]_
+As a result, the second last major dependency on distutils is gone and
+there is no need to keep it in the standard library.
+
+The final dependency on distutils is CPython itself, which uses it to
+build the native extension modules in the standard library (except on
+Windows). Because this is a CPython build-time dependency, it is
+possible to continue to use distutils for this specific case without
+it being part of the standard library.
+
+Deprecation and removal will make it obvious that issues should be
+fixed in the setuptools project, and will reduce a source of bug
+reports and test maintenance that is unnecessary. It will also help
+promote the development of alternative build backends, which can now
+be supported more easily thanks to PEP 517.
+
+
+Specification
+=============
+
+In Python 3.10 and 3.11, distutils will be formally marked as
+deprecated. All known issues will be closed at this time.
+``import distutils`` will raise a deprecation warning.
+
+During Python 3.10 and 3.11, uses of distutils within the standard
+library may change to use alternative APIs.
+
+In Python 3.12, distutils will no longer be installed by ``make
+install`` or any of the first-party distribution. Third-party
+redistributors should no longer include distutils in their bundles or
+repositories.
+
+This PEP makes no specification on migrating the parts of the CPython
+build process that currently use distutils. Depending on
+contributions, this migration may occur at any time.
+
+After Python 3.12 is started and when the CPython build process no
+longer depends on distutils being in the standard library, the entire
+``Lib/distutils`` directory and ``Lib/test/test_distutils.py`` file
+will be removed from the repository.
+
+Other references to distutils will be cleaned up. As of Python 3.9's
+initial release, the following modules have references in code or
+comments:
+
+* Lib/ctypes/util.py
+* Lib/site.py
+* Lib/sysconfig.py
+* Lib/_aix_support.py
+* Lib/_bootsubprocess.py
+* Lib/_osx_support.py
+* Modules/_decimal/tests/formathelper.py
+
+As the distutils code is already included in setuptools, there is no
+need to republish it in any other form. Those who require access to
+the functionality should use setuptools or an alternative build
+backend.
+
+Backwards Compatibility
+=======================
+
+Code that imports distutils will no longer work from Python 3.12.
+
+The suggested migration path is to use the equivalent (though not
+identical) imports from setuptools.
+
+Compatibility shims already exist in setuptools and pip to
+transparently migrate old code, however, these are controlled by
+external projects and are in no way bound by this PEP. Consult their
+latest documentation for migration advice.
+
+Some projects use alternate sets of shims over distutils, notably,
+numpy.distutils. [4]_ Where known, these projects have been informed.
```

Comment:
Do you mean https://github.com/pypa/setuptools/issues/2372 ? "Have been informed" was only true immediately as I wrote it, because I interrupted typing at this point to make sure it had been mentioned ;)

Reply:
Thanks, I have now subscribed to that discussion.